lib/opsgenie_alert_tagging.rb

```diff
@@ -21,4 +25,54 @@ def list_all_alerts(date)
 
     results["data"]
   end
+
+  def all_updated_alerts
+    all_alerts_for_specific_date = list_all_alerts
+    results = []
+    all_alerts_for_specific_date.each do |raw_alert|
+
+      alert = OpsgenieAlert.new(raw_alert)
+      result = {}
+      result[:tags] = alert.tags
+      result[:id] = alert.id
+
+      if alert.bank_holiday?
+        result[:tags] << "wakinghours"
+        result[:tags] << "OOH"
+
+      elsif alert.inhours? && !(alert.saturday? || alert.sunday?)
+        result[:tags] << "inhours"
+
+      elsif alert.wakinghours_weekday?
+        result[:tags] << "wakinghours"
+        result[:tags] << "OOH"
+
+      elsif alert.wakinghours_weekend? && (alert.saturday? || alert.sunday?)
+        result[:tags] << "wakinghours"
+        result[:tags] << "OOH"
+
+      elsif alert.sleepinghours?
+        result[:tags] << "sleepinghours"
+        result[:tags] << "OOH"
+      end
+
+      result[:tags].uniq!
+      results << result
+    end
+    results
+  end
```

Comment:
Making the changes above, this could be:

```suggestion
  def all_updated_alerts
    all_alerts_for_specific_date = list_all_alerts
    
    all_alerts_for_specific_date.map do |raw_alert|
      alert = OpsgenieAlert.new(raw_alert)
      
      result = {}
      result[:id] = alert.id
      
      result[:tags] = alert.tags
      result[:tags] << "inhours" if alert.in_hours?
      result[:tags] << "OOH" if alert.out_of_hours?
      result[:tags] << "wakinghours" if alert.during_waking_hours?
      result[:tags] << "sleepinghours" if alert.during_sleeping_hours?
      
      result
    end
  end
```

You could then even make `OpsgenieAlert#tags` do the extra tagging for you.